R/tellTS.R

```diff
@@ -1,13 +1,145 @@
+#' Model Behavior." Water Resources Research 49 (3): 1400-1414.
+#' doi: \href{http://dx.doi.org/10.1002/wrcr.20124}{10.1002/wrcr.20124}
+#' @keywords models
+#' @examples
+#'
+#' library(sensitivity)
+#'
+#' ## Load data
+#' data(Cotter)
+#' obs <- Cotter[1:1000]
+#'
+#' ## Define rainfall-runoff model structure
+#' model.str <- hydromad(obs,
+#'   sma = "cwi", routing = "expuh",
+#'   tau_q = c(0, 2), tau_s = c(2, 100), v_s = c(0, 1)
+#' )
+#'
+#' ## Set the random seed to obtain replicable results
+#' set.seed(19)
+#'
+#' ## Setup Morris Sensitivity analysis
+#' incomplete <- morris(
+#'   ## Names of factors/parameters
+#'   factors = names(getFreeParsRanges(model.str)),
+#'   ## Number of repetitions
+#'   r = 4,
+#'   ## List specifying design type and its parameters
+#'   design = list(type = "oat", levels = 10, grid.jump = 2),
+#'   ## Minimum value of each non-fixed parameter
+#'   binf = sapply(getFreeParsRanges(model.str), min),
+#'   ## Maximum value of each non-fixed parameter
+#'   bsup = sapply(getFreeParsRanges(model.str), max)
+#' )
+#'
+#'
+#' # Calculate rolling time series of objective for each parameter set,
+#' #   keeping results in memory
+#' runs <- evalParsRollapply(incomplete$X, model.str,
+#'   objective = ~ hmadstat("r.squared")(Q, X) /
+#'     (2 - hmadstat("r.squared")(Q, X)),
+#'   parallel = "none",
+#'   filehash.name = NULL
+#' )
+#'
+#' # Calculate Morris elementary effects for each timestep
+#' sens <- tellTS(incomplete, runs, parallel = "none")
+#'
+#' ## Plot time series of mean absolute elementary effects
+#' library(ggplot2)
+#' ggplot2::qplot(x = ts.id, y = mu.star, colour = variable, data = sens, geom = "line")
+#'
+#' ## And of standard deviation of elementary effects
+#' ggplot2::qplot(x = ts.id, y = sigma, colour = variable, data = sens, geom = "line")
+#'
+#' ######################################################################
+#' \dontrun{
+#' ## Sobol indices
+#' ### Setup parallelisation
+#' library(parallel)
+#' hydromad.options("parallel" = list("evalParsRollapply" = "clusterApply"))
+#' hydromad.options("parallel" = list("tellTS" = "clusterApply"))
+#' cl <- makeCluster(3)
+#' clusterEvalQ(cl, library(hydromad))
+#'
+#' ### Create sobol design
+#' n <- 1000 ## Set number of samples desired
+#' X1 <- parameterSets(getFreeParsRanges(model.str), n)
+#' X2 <- parameterSets(getFreeParsRanges(model.str), n)
+#' incomplete <- sobol2002(model = NULL, X1 = X1, X2 = X2, nboot = 100)
+#'
+#' ### Run model for each parameter set, storing result to disk
+#' system.time(runs <- evalParsRollapply(incomplete$X, model.str,
+#'   objective = hmadstat("r.squared"), parallel = "clusterApply"
+#' ))
+#'
+#' ## Calculate sobol indices for each timestep with stored runs
+#' system.time(TSI <- tellTS(incomplete, runs))
+#'
+#' ## Plot time series of sensitivities
+#' library(ggplot2)
+#' ggplot2::qplot(x = ts.id, y = original, colour = variable, data = TSI, geom = "line")
```

Comment:
If we use library(ggplot2) above, then we don't need to use ggplot2:: 
This is an example, so I'd be inclined to using only library. Do you think that's the right solution?

Reply:
I've removed `ggplot::` from these examples for now, but I'm pretty sure I originally put them there because errors would pop up without them. It seems to be working locally, so I'll also check whether the removal affects the CI at all - if not, then we should be fine.

Reply:
Thanks!

Reply:
Seems to be fine, no complaints from R CMD check!